core/src/main/scala/kafka/message/InvalidMessageException.scala

```diff
@@ -17,9 +17,11 @@
 
 package kafka.message
 
+import org.apache.kafka.common.errors.CorruptRecordException
+
 /**
  * Indicates that a message failed its checksum and is corrupt
  */
-class InvalidMessageException(message: String) extends RuntimeException(message) {
+class InvalidMessageException(message: String) extends CorruptRecordException(message) {
```

Comment:
It may be worth adding a comment explaining this as it's not obvious to someone reading.